lib/auth.ts

```diff
@@ -450,41 +450,120 @@ const getAuth = function (
 	 *
-	 * @fulfil {String} - session token
+	 * @fulfil {(String|undefined)} - session token, if email verification is not enabled; else, undefined
 	 * @returns {Promise}
 	 *
 	 * @example
 	 * balena.auth.register({
 	 * 	email: 'johndoe@gmail.com',
 	 * 	password: 'secret'
-	 * }).then(function(token) {
-	 * 	console.log(token);
+	 * }).then(function(result) {
+	 * 	if (result) {
+	 * 	  console.log(`Token: ${result}`);
+	 * 	} else {
+	 *    console.log('Registration in-process with email verification pending!');
+	 *  }
+	 * }).catch(function(error) {
+	 * 	throw error;
 	 * });
 	 *
-	 * @example
-	 * balena.auth.register({
-	 * 	email: 'johndoe@gmail.com',
-	 * 	password: 'secret'
-	 * }, function(error, token) {
-	 * 	if (error) throw error;
-	 * 	console.log(token);
-	 * });
 	 */
 	async function register(credentials: {
 		email: string;
 		password: string;
 		'g-recaptcha-response'?: string;
-	}): Promise<string> {
+	}): Promise<string | undefined> {
 		const { body } = await request.send({
 			method: 'POST',
 			url: '/user/register',
 			baseUrl: apiUrl,
 			body: credentials,
 			sendToken: false,
 		});
+
+		// With email verification enabled we return with status(201) i.e. no auto-login
+		if (body === 'Created') {
+			return;
+		}
+
 		return body;
 	}
 
+	/**
+	 * @summary Verifies an email
+	 * @name verifyEmail
+	 * @public
+	 * @function
+	 * @memberof balena.auth
+	 *
+	 * @param {Object} verificationPayload - in the form of email, and token
+	 * @param {String} verificationPayload.email - the email
+	 * @param {String} verificationPayload.token - the verification token
+	 *
+	 * @fulfil {String} - session token
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.auth.verifyEmail({
+	 * 	email: 'johndoe@gmail.com',
+	 * 	token: '5bb11d90eefb34a70318f06a43ef063f'
+	 * }).then(function(jwt) {
+	 * 	console.log(jwt);
+	 * });
+	 *
+	 */
+	async function verifyEmail(verificationPayload: {
+		email: string;
+		token: string;
+	}): Promise<string> {
+		const email = verificationPayload.email;
+		const verificationToken = verificationPayload.token;
+
+		const { body } = await request.send({
+			method: 'POST',
+			url: '/user/v1/verify-email',
+			body: {
+				verificationToken,
+				email,
+			},
+			baseUrl: apiUrl,
+			sendToken: false,
+		});
+
+		return body;
+	}
+
+	/**
+	 * @summary Re-send verification email to the user
+	 * @name requestVerificationEmail
+	 * @public
+	 * @function
+	 * @memberof balena.auth
+	 * @description This will only work if you used {@link balena.auth.login} to log in.
+	 *
+	 * @fulfil {string} - returns with 'OK' status when successful
+	 * @returns {Promise}
+	 *
+	 * @example
+	 * balena.auth.requestVerificationEmail().then(function() {
+	 * 	console.log('Requesting verification email operation complete!');
+	 * }).catch(function(error) {
+	 * 	if (error) throw error;
+	 * 	console.log(email);
+	 * });
+	 *
+	 */
+	async function requestVerificationEmail() {
```

Comment:
How about v ?
```suggestion
	async function resendVerificationEmail() {
```
I see a point though of this naming for pre-existing users that later try to verify their account, since "resend" doesn't make much sense if you have never been sent one before...

Reply:
Confused here - do you prefer the "resend" over "request"? From your comment doesn't look like it; but you change shows me that suggested.

Reply:
I'm in favor of my code change suggestion. The rest were just thoughts.

Reply:
As you also mentioned lets keep the `requestVerificationEmail()` name given we are using this for pre-existing users as well.